### PR TITLE
feat/has-pipe

### PR DESCRIPTION
**Pull request by Lightllord**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/cdk/integration/tests/jest/pipes/has-pipr.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { TestBed } from '@angular/core/testing';
+import { HasPipe, HasPipeModule } from '@angular-ru/cdk/pipes';
+
+describe('deep path', () => {
```
Comment (Schirbak) [inline packages/cdk/integration/tests/jest/pipes/has-pipr.spec.ts:4]: correct this

File: packages/cdk/pipes/src/has/has.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+@Pipe({
+    name: 'has'
+})
+export class HasPipe implements PipeTransform {
+    public transform<T>(set: Nullable<Set<T>>, entry: T): boolean {
+        if (isNil(set)) {
+            return false;
+        } else {
+            return set.has(entry);
```
Comment (Schirbak) [inline packages/cdk/pipes/src/has/has.pipe.ts:13]: simplify: 
      `return set?.has(entry) ?? false`

File: packages/cdk/pipes/src/includes/includes.pipe.ts [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
-    public transform<T>(array: Nullable<T[]>, entry: T): boolean {
-        return array?.includes(entry) ?? false;
+    public transform<T>(array: Nullable<T[] | Set<T>>, entry: T): boolean {
+        if (isNil(array)) {
+            return false;
+        } else {
+            const forCheck: Set<T> = new Set(array);
+
+            return forCheck.has(entry);
+        }
```
Comment (Schirbak) [inline packages/cdk/pipes/src/includes/includes.pipe.ts:14]: revert

**Review by Schirbak — CHANGES_REQUESTED** [review]

File: packages/cdk/integration/tests/jest/pipes/has-pipr.spec.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { TestBed } from '@angular/core/testing';
```
[the remark this answers is not in the document]
Comment (Schirbak) [reply packages/cdk/integration/tests/jest/pipes/has-pipr.spec.ts:1]: it's a shame that the spell checker doesn't check filenames :pepe:

**Review by splincode — APPROVED** [review]

**Pull request merged**